### PR TITLE
Include duplicate package name in error message

### PR DESCRIPTION
**Pull request by athomas**

### CODE REVIEW

File: lib/src/package_config_impl.dart [hunk 1]
```diff
@@ -77,7 +77,7 @@ class SimplePackageConfig implements PackageConfig {
       var name = package.name;
       if (packageNames.contains(name)) {
         onError(PackageConfigArgumentError(
-            name, "packages", "Duplicate package name"));
+            name, "packages", "Duplicate package $name"));
```
Comment (jakemac53) [inline lib/src/package_config_impl.dart]: Should be part of the `toString` of the `PackageConfigArgumentError` error generally?

Comment (lrhn) [reply lib/src/package_config_impl.dart]: Not sure what you are asking. Answer is probably "no" 😉 .

The `PackageConfigArgumentError` is just an `ArgumentError` which implements `PackageConfigError` (just so you can catch all `PackageConfigError`s).
It doesn't have specific fields for "package name errors".

Here it's just an argument error, the argument in error is `packages`, the problem description is that `packages` contains two entries with the same package name (now `$name`).

Comment (lrhn) [inline lib/src/package_config_impl.dart]: I'd do `"Duplicate package name '$name'"`. It's not a duplicate package (well, maybe it is, maybe it isn't, but let's not assume).

**Review by lrhn — APPROVED** [review]

File: lib/src/package_config_impl.dart (hunk 1 above)
Comment (athomas) [reply lib/src/package_config_impl.dart]: Done.

**Pull request merged**